substrate/frame/examples/basic/src/lib.rs

```diff
@@ -177,267 +188,199 @@ pub mod pallet {
 			Weight::zero()
 		}
 
-		// `on_finalize` is executed at the end of block after all extrinsic are dispatched.
+		// `on_finalize` is executed at the end of block after all extrinsics are dispatched.
 		fn on_finalize(_n: BlockNumberFor<T>) {
 			// Perform necessary data/state clean up here.
 		}
+	}
 
-		// A runtime code run after every block and have access to extended set of APIs.
-		//
-		// For instance you can generate extrinsics for the upcoming produced block.
-		fn offchain_worker(_n: BlockNumberFor<T>) {
-			// We don't do anything here.
-			// but we could dispatch extrinsic (transaction/unsigned/inherent) using
-			// sp_io::submit_extrinsic.
-			// To see example on offchain worker, please refer to example-offchain-worker pallet
-			// accompanied in this repository.
+	/// An example storage item to store a single value, in our case, some Balance.
+	/// This storage item uses [`OptionQuery`] by default which will return what is in the actual DB
+	/// state. If this is `None`, then the resulting query will be `None`.
+	///
+	/// The getter attribute generates a function on the `Pallet` struct that we can use to
+	/// conveniently retrieve the current value stored.
+	#[pallet::storage]
+	#[pallet::getter(fn dummy)]
+	pub(super) type Dummy<T: Config> = StorageValue<_, T::Balance>;
+
+	/// An example storage item that stores a `u32` value.
+	/// Here, we demonstrate using [`ValueQuery`] instead of the default query which will always
+	/// return a value.
+	///
+	/// In our example, if ever the stored value is removed, the value stored will be `0` which is
+	/// `u32::default()`.
+	#[pallet::storage]
+	#[pallet::getter(fn dummy_value_query)]
+	pub(super) type DummyValueQuery<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	/// An example storage map that has enumerable entries. In our case this is a mapping of
+	/// AccountIds to Balances.
+	///
+	/// We don't actually use this anywhere.
+	#[pallet::storage]
+	#[pallet::getter(fn accounts_to_balances_map)]
+	pub(super) type AccountsToBalancesMap<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, T::Balance>;
+
+	/// The genesis configuration type.
+	#[pallet::genesis_config]
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config> {
+		pub dummy: T::Balance,
+		pub dummy_value_query: u32,
+		pub accounts_to_balances_map: Vec<(T::AccountId, T::Balance)>,
+	}
+
+	/// The genesis build for this pallet.
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			Dummy::<T>::put(&self.dummy);
+			DummyValueQuery::<T>::put(&self.dummy_value_query);
+			for (a, b) in &self.accounts_to_balances_map {
+				AccountsToBalancesMap::<T>::insert(a, b);
+			}
 		}
 	}
 
-	// The call declaration. This states the entry points that we handle. The
+	/// Events are a simple means of reporting specific conditions and
+	/// circumstances that have happened that users, Dapps and/or chain explorers would find
```

Comment:
have you see https://polkadot-blockchain-academy.github.io/pba-book/frame/events-errors/slides.html#/13? could be a good input.

Reply:
although again, if you want to improve this further, it better go into the docs for `pallet::event` macro stub.